auth/oidc.go

```diff
@@ -242,25 +246,34 @@ func (opm OIDCProviderMap) Stop() {
 	}
 }
 
-func (op *OIDCProvider) GetClient(buildCallbackURLFunc OIDCCallbackURLFunc) *OIDCClient {
-	// Initialize the client on first request. If the callback URL isn't defined for the provider,
-	// uses buildCallbackURLFunc to construct (based on current request)
-	op.clientOnce.Do(func() {
-		var err error
-		// If the redirect URL is not defined for the provider generate it from the
-		// handler request and set it on the provider
-		if op.CallbackURL == nil || *op.CallbackURL == "" {
-			callbackURL := buildCallbackURLFunc(op.Name, op.IsDefault)
-			if callbackURL != "" {
-				op.CallbackURL = &callbackURL
-			}
-		}
-		if err = op.initOIDCClient(); err != nil {
-			base.Errorf("Unable to initialize OIDC client: %v", err)
-		}
-	})
+// GetClient initializes the client on first successful request.
+func (op *OIDCProvider) GetClient(buildCallbackURLFunc OIDCCallbackURLFunc) (*OIDCClient, error) {
+	// If the callback URL isn't defined for the provider, uses buildCallbackURLFunc to construct (based on current request)
+
+	if op.clientInit.IsTrue() { // If client not initialized
```

Comment:
Comment is outdated but probably not required

```suggestion
	if op.clientInit.IsTrue() {
```